packages/web-app-files/src/components/SideBar/Versions/FileVersions.vue

```diff
@@ -92,21 +92,22 @@ export default {
       await this.loadVersions({ client: this.$client, fileId: this.highlightedFile.fileId })
       this.loading = false
     },
-    revertVersion(file) {
-      this.$client.fileVersions
-        .restoreFileVersion(
-          this.highlightedFile.id,
-          this.currentVersionId(file),
-          this.highlightedFile.path
-        )
-        .then(async () => {
-          const resource = await (this.$clientService.webdav as WebDAV).getFileInfo(
-            this.space,
-            this.highlightedFile
-          )
-          this.UPDATE_RESOURCE(resource)
-          return this.fetchFileVersions()
-        })
+    async revertVersion(file) {
+      const { fileId, path } = this.highlightedFile
+      await this.$client.fileVersions.restoreFileVersion(fileId, this.currentVersionId(file), path)
+      const resource = await (this.$clientService.webdav as WebDAV).getFileInfo(
+        this.space,
+        this.highlightedFile
+      )
+
+      const fieldsToUpdate = ['size', 'mdate']
+      for (const field of fieldsToUpdate) {
+        if (this.highlightedFile[field]) {
+          this.UPDATE_RESOURCE_FIELD({ id: fileId, field, value: resource[field] })
```

Comment:
When being on e.g. the `Shared with me` page the store holds `sharedResource` entities. Since the `UPDATE_SOURCE_FIELD` mutation identifies entities by the `id` prop it looks wrong to me that you're using the `fileId` here. How does it actually work at all if you're not in a personal- or project-space (where id=fileId)? 😅

Reply:
Good catch, the need the `id` here indeed. I didn't notice it because the share tables don't have the columns "size" and "mdate" anyway.